File: Algorithms/src/762. Prime Number of Set Bits in Binary Representation/Code/solution.js
/**
 * @param {number} left
 * @param {number} right
 * @return {number}
 */
var countPrimeSetBits = function (left, right) {
  const primeSet = new Set([2, 3, 5, 7, 11, 13, 17, 19]);

  let ans = 0;

  for (let num = left; num <= right; num++) {
    // Convert to binary and count 1s
    let setBits = num.toString(2).split("1").length - 1;

    if (primeSet.has(setBits)) {
      ans++;
    }
  }

  return ans;
};

File: Algorithms/src/762. Prime Number of Set Bits in Binary Representation/Code/solution.go
import "math/bits"

func countPrimeSetBits(left int, right int) int {
    
    primeSet := map[int]bool{
        2:true, 3:true, 5:true, 7:true,
        11:true, 13:true, 17:true, 19:true,
    }
    
    ans := 0
    
    for num := left; num <= right; num++ {
        // Count set bits
        setBits := bits.OnesCount(uint(num))
        
        if primeSet[setBits] {
            ans++
        }
    }
    
    return ans
}